lotus-soup/testkit/role_miner.go

```diff
@@ -353,86 +355,51 @@ func (m *LotusMiner) RunDefault() error {
 	var (
 		t       = m.t
 		clients = t.IntParam("clients")
-		miners  = t.IntParam("miners")
 	)
 
 	t.RecordMessage("running miner")
-	t.RecordMessage("block delay: %v", build.BlockDelaySecs)
 	t.D().Gauge("miner.block-delay").Update(float64(build.BlockDelaySecs))
 
 	ctx := context.Background()
-	myActorAddr, err := m.MinerApi.ActorAddress(ctx)
-	if err != nil {
-		return err
-	}
 
 	// mine / stop mining
-	mine := true
 	done := make(chan struct{})
 
-	if m.MineOne != nil {
-		go func() {
-			defer t.RecordMessage("shutting down mining")
-			defer close(done)
-
-			var i int
-			for i = 0; mine; i++ {
-				// synchronize all miners to mine the next block
-				t.RecordMessage("synchronizing all miners to mine next block [%d]", i)
-				stateMineNext := sync.State(fmt.Sprintf("mine-block-%d", i))
-				t.SyncClient.MustSignalAndWait(ctx, stateMineNext, miners)
-
-				ch := make(chan error)
-				const maxRetries = 100
-				success := false
-				for retries := 0; retries < maxRetries; retries++ {
-					err := m.MineOne(ctx, func(mined bool, err error) {
-						if mined {
-							t.D().Counter(fmt.Sprintf("block.mine,miner=%s", myActorAddr)).Inc(1)
-						}
-						ch <- err
-					})
-					if err != nil {
-						panic(err)
-					}
-
-					miningErr := <-ch
-					if miningErr == nil {
-						success = true
-						break
-					}
-					t.D().Counter("block.mine.err").Inc(1)
-					t.RecordMessage("retrying block [%d] after %d attempts due to mining error: %s",
-						i, retries, miningErr)
-				}
-				if !success {
-					panic(fmt.Errorf("failed to mine block %d after %d retries", i, maxRetries))
-				}
-			}
+	if t.StringParam("mining_mode") != "natural" {
+		gen, err := m.FullApi.ChainGetGenesis(context.Background())
+		if err != nil {
+			return fmt.Errorf("failed to get genesis: %w", err)
+		}
+
+		var (
+			genTime             = time.Unix(int64(gen.MinTimestamp()), 0)
+			localEpochAdvanceCh = make(chan abi.ChainEpoch, 128)
+			globalEpochStartCh  = make(chan abi.ChainEpoch, 128)
```

Comment:
I may be reading this wrong, it's a bit confusing. There seems to be some discrepancy in the names here and in the functions arguments, like some of the `Start` should be `Advance`. (This might be intended and I'm missing something, in that case could you add some extra clarifications please?)